lib/kafo/progress_bar.rb

```diff
@@ -42,15 +42,28 @@ def update(line)
       end
 
       if (line_start = EVALTRACE_START.match(line))
-        if (known_resource = find_known_resource(line_start[1]))
+        if line_start[4]
+          # Puppet 6.6 introduced progress in evaltrace
+          # Puppet counts 1-based where we count 0-based here
+          new_lines = line_start[3].to_i - 1
+          new_total = line_start[4].to_i
+          if new_lines != @lines || @total != new_total
+            @lines = new_lines
+            @total = new_total
+            update_bar = true
+            force_update = true
+          end
+        end
+
+        if (known_resource = find_resource(line_start[1]))
```

Comment:
With Puppet 6, we no longer hook in internals. This part never runs:

https://github.com/theforeman/kafo/blob/6c7f49ff4e668853ca25ff07f1d4dd94ca7e9602/modules/kafo_configure/lib/puppet/parser/functions/add_progress.rb#L28

That means `MONITOR_RESOURCE` is a Puppet < 6 thing and `@resources` is never filled. So if you see `line_start[4]`, you know it's Puppet 6.6 and `find_known_resource` will never work.

This is the part I struggled with.

I think this should be an `elsif` and set `line` in the `if line_start[4]` part.

Reply:
Note how I changed this find_resource and re-factored the method to actually derive the resource. This does still feel nice so that you get the output as:

```
Notify[test]
Prefetching example resources for example_type 
File[/foo/bar]
```

Without deriving the resource as we had before I believe we end up with the reporting more like:

```
/Stage[main]/Example/Notify[test]
Prefetching example resources for example_type
/Stage[main]/Example/File[/foo/bar]
```

This is more verbose of output, and from a progress bar standpoint, I would think someone would only care to mentally parse, when reading, the way we do it today (e.g. File[/foo/bar]) rather than the full path. I think the other benefit is that it shows the most relevant info on the screen, for example:

```
Installing             Info: /Package[foreman-dynflow-sidekiq]: Starting  [6%] [.                       ]
```

Which, if it was the full path, I think would truncate the end off and user would not see the most important part. Granted, most of this information flies off the screen.